pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/NullHandlingIntegrationTest.java

```diff
@@ -459,4 +459,29 @@ public Object[][] nullLiteralQueries() {
         {String.format("SELECT tan(null) FROM %s", getTableName()), "null"}
     };
   }
+
+  /// This test ensures IS_TRUE can be trimmed off on leaf stage
+  @Test(dataProvider = "useBothQueryEngines")
+  public void testFilteredAggregationNoScanInFilter(boolean useMultiStageQueryEngine)
+      throws Exception {
+    setUseMultiStageQueryEngine(useMultiStageQueryEngine);
+
+    String query = "SELECT city, COUNT(*), COUNT(*) FILTER(WHERE description = 'unknown') FROM mytable GROUP BY city";
+
+    // MSE will insert IS_TRUE to the aggregate filter
+    if (useMultiStageQueryEngine) {
+      explainLogical(query,
+          "Execution Plan\n"
+              + "PinotLogicalAggregate(group=[{0}], agg#0=[COUNT($1)], agg#1=[COUNT($2)], aggType=[FINAL])\n"
+              + "  PinotLogicalExchange(distribution=[hash[0]])\n"
+              + "    PinotLogicalAggregate(group=[{0}], agg#0=[COUNT()], agg#1=[COUNT() FILTER $1], aggType=[LEAF])\n"
+              + "      LogicalProject(city=[$5], $f1=[IS TRUE(=($7, _UTF-8'unknown'))])\n"
```

Comment:
So this `IS_TRUE` is inserted in order to filter out null values because `=(null, 'unknown')` is `null` and `IS_TRUE(null)` evaluates to `false`?

Reply:
And we don't want to preserve the `IS_TRUE` predicate because our implementation of `=` doesn't return `null`?

Reply:
> And we don't want to preserve the IS_TRUE predicate because our implementation of = doesn't return null?

`IS_TRUE(x)` never returns null (in SQL). It is semantically equal to `x = true and x is not null`, which always evaluate to true or false.

I think this IS_TRUE is added by Calcite to filter out nulls on the predicate, as it should be evaluated to true or false. Am I right Jackie?

Reply:
Yep, I'm asking about the EQUALS implementation, not IS_TRUE 😄

Reply:
Pinot leaf stage (SSE) is able to handle this within filter clause. There are 3 possible values for predicate: `true`, `false`, `null` (unknown). SSE will keep records evaluated to `true` only, thus this `IS_TRUE` is redundant

`EQUALS` can evaluate to `null` when either the left hand side or right hand side is `null`